File: frontend/src/types/listing.ts
export type Platform = 'AMAZON' | 'FLIPKART' | 'MEESHO' | 'SHOPIFY';
export type ListingStatus = 'DRAFT' | 'PUBLISHED' | 'ARCHIVED';

export interface Listing {
  id: number;
  productName: string;
  productDescription: string | null;
  category: string | null;
  brand: string | null;
  material: string | null;
  color: string | null;
  size: string | null;
  imageUrl: string | null;
  originalFileName: string | null;
  platform: Platform;
  seoTitle: string | null;
  bulletPoints: string | null;
  description: string | null;
  tags: string | null;
  keywords: string | null;
  metaDescription: string | null;
  platformFormattedListing: string | null;
  status: ListingStatus;
  createdAt: string;
  updatedAt: string;
}

export interface ListingRequest {
  productName: string;
  productDescription?: string;
  category?: string;
  brand?: string;
  material?: string;
  color?: string;
  size?: string;
  platform: Platform;
}

export interface ListingQueryParams {
  page?: number;
  size?: number;
  sortBy?: string;
  sortDir?: 'asc' | 'desc';
  keyword?: string;
  platform?: Platform;
  status?: ListingStatus;
}

export interface ListingStats {
  totalListings: number;
  draftCount: number;
  publishedCount: number;
  archivedCount: number;
  platformStats: Record<Platform, number>;
}

File: src/main/java/com/ailisting/model/entity/Listing.java
package com.ailisting.model.entity;

import com.ailisting.model.enums.Platform;
import jakarta.persistence.*;
import lombok.*;
import org.hibernate.annotations.CreationTimestamp;
import org.hibernate.annotations.UpdateTimestamp;

import java.time.LocalDateTime;

@Entity
@Table(name = "listings")
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class Listing {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "user_id", nullable = false)
    private User user;

    @Column(name = "product_name", nullable = false, length = 255)
    private String productName;

    @Column(name = "product_description", columnDefinition = "TEXT")
    private String productDescription;

    @Column(name = "category", length = 100)
    private String category;

    @Column(name = "brand", length = 100)
    private String brand;

    @Column(name = "material", length = 100)
    private String material;

    @Column(name = "color", length = 100)
    private String color;

    @Column(name = "size", length = 100)
    private String size;

    @Column(name = "image_url", length = 500)
    private String imageUrl;

    @Column(name = "original_file_name", length = 500)
    private String originalFileName;

    @Enumerated(EnumType.STRING)
    @Column(name = "platform", nullable = false, length = 20)
    private Platform platform;

    @Column(name = "seo_title", length = 500)
    private String seoTitle;

    @Column(name = "bullet_points", columnDefinition = "TEXT")
    private String bulletPoints;

    @Column(name = "description", columnDefinition = "TEXT")
    private String description;

    @Column(name = "tags", columnDefinition = "TEXT")
    private String tags;

    @Column(name = "keywords", length = 500)
    private String keywords;

    @Column(name = "meta_description", length = 300)
    private String metaDescription;

    @Column(name = "platform_formatted_listing", columnDefinition = "TEXT")
    private String platformFormattedListing;

    @Enumerated(EnumType.STRING)
    @Column(name = "status", nullable = false, length = 20)
    @Builder.Default
    private ListingStatus status = ListingStatus.DRAFT;

    @Column(name = "deleted", nullable = false)
    @Builder.Default
    private boolean deleted = false;

    @CreationTimestamp
    @Column(name = "created_at", updatable = false)
    private LocalDateTime createdAt;

    @UpdateTimestamp
    @Column(name = "updated_at")
    private LocalDateTime updatedAt;

    public enum ListingStatus {
        DRAFT, PUBLISHED, ARCHIVED
    }
}